ayon_api/server_api.py

```diff
@@ -1674,6 +1669,53 @@ def download_file(
             progress.set_transfer_done()
         return progress
 
+    def download_file(
+        self, endpoint, filepath, chunk_size=None, progress=None
+    ):
+        """Download file from AYON server.
+
+        Endpoint can be full url (must start with 'base_url' of api object).
+
+        Progress object can be used to track download. Can be used when
+        download happens in thread and other thread want to catch changes over
+        time.
+
+        Todos:
+            Use retries and timeout.
+            Return RestApiResponse.
+
+        Args:
+            endpoint (str): Endpoint or URL to file that should be downloaded.
+            filepath (str): Path where file will be downloaded.
+            chunk_size (Optional[int]): Size of chunks that are received
+                in single loop.
+            progress (Optional[TransferProgress]): Object that gives ability
+                to track download progress.
+
+        """
+        # Create dummy object so the function does not have to check
+        #   'progress' variable everywhere
+        if progress is None:
+            progress = TransferProgress()
+
+        progress.set_destination_url(filepath)
+
+        dst_directory = os.path.dirname(filepath)
+        if not os.path.exists(dst_directory):
+            os.makedirs(dst_directory, exist_ok=True)
```

Comment:
When using `exist_ok` there's no need to check for existence?
```suggestion
         os.makedirs(dst_directory, exist_ok=True)
```